lsp/nls/src/requests/completion.rs

```diff
@@ -1,16 +1,198 @@
 use codespan::ByteIndex;
 use codespan_lsp::position_to_byte_index;
+use lazy_static::lazy_static;
 use log::debug;
-use lsp_server::{RequestId, Response, ResponseError};
+use lsp_server::{ErrorCode, RequestId, Response, ResponseError};
 use lsp_types::{CompletionItem, CompletionParams};
+use nickel_lang::{
+    identifier::Ident,
+    term::{Contract, MetaValue},
+    types::{AbsType, RowIteratorItem, Types},
+};
 use serde_json::Value;
 
 use crate::{
-    linearization::interface::TermKind,
-    server::Server,
+    linearization::{
+        completed::Completed,
+        interface::{TermKind, UsageState, ValueState},
+        LinearizationItem,
+    },
+    server::{self, Server},
     trace::{Enrich, Trace},
 };
 
+/// Find record fields for an item with the specified id.
+fn find_record_fields(
+    linearization: &Vec<LinearizationItem<Types>>,
+    id: usize,
+) -> Option<Vec<Ident>> {
+    linearization.iter().find_map(|item| match item.kind {
```

Comment:
I may be missing something again, but we still do a linear search through the whole linearization, don't we? The point of indices is to have constant time access. Couldn't we rather use `Completed::get_item` instead of doing this `iter()`?

Reply:
Yes. Nice catch, thanks!